pancancer_evaluation/mutation_prediction.py

```diff
@@ -0,0 +1,465 @@
+        debug (bool): if True, use a subset of expression data for quick debugging
+        """
+        # save relevant parameters
+        np.random.seed(seed)
+        self.seed = seed
+        self.results_dir = results_dir
+        self.subset_mad_genes = subset_mad_genes
+        self.verbose = verbose
+
+        # load and store data in memory
+        self._load_data(debug=debug)
+
+
+    def load_gene_set(self, gene_set='top_50'):
+        """
+        Load gene set data from previous GitHub repos.
+
+        Arguments
+        ---------
+        gene_set (str): which predefined gene set to use, or 'custom' for custom list.
+
+        Returns
+        -------
+        genes_df (pd.DataFrame): list of genes to run cross-validation experiments for,
+                                 contains gene names and oncogene/TSG classifications
+
+        TODO: still not sure how to generalize oncogene/TSG info past these
+        predefined gene sets, should eventually look into how to do this
+        """
+        if self.verbose:
+            print('Loading gene label data...', file=sys.stderr)
+
+        if gene_set == 'top_50':
+            genes_df = du.load_top_50()
+        elif gene_set == 'vogelstein':
+            genes_df = du.load_vogelstein()
+        else:
+            assert isinstance(gene_set, typing.List)
+            genes_df = du.load_top_50()
+            genes_df = genes_df[genes_df['gene'].isin(gene_set)]
+
+        return genes_df
+
+
+    def process_data_for_gene(self,
+                              gene,
+                              classification,
+                              use_pancancer=False,
+                              shuffle_labels=False):
+        """
+        Prepare to run cancer type experiments for a given gene.
+
+        This has to be rerun for each gene, since the data is filtered based
+        on label proportions for the given gene in each cancer type.
+
+        Arguments
+        ---------
+        gene (str): gene to run experiments for
+        classification (str): 'oncogene' or 'TSG'; most likely cancer function for
+                              the given gene
+        use_pancancer (bool): whether or not to use pancancer data
+        shuffle_labels (bool): whether or not to shuffle labels (negative control)
+        """
+        self._make_gene_dir(gene, use_pancancer)
+
+        y_df_raw = self._generate_labels(gene, classification)
+
+        filtered_data = self._filter_data_for_gene(
+            self.rnaseq_df,
+            y_df_raw,
+            use_pancancer
+        )
+        rnaseq_filtered_df, y_filtered_df, gene_features = filtered_data
+
+        if shuffle_labels:
+            y_filtered_df.status = np.random.permutation(
+                y_filtered_df.status.values)
+
+        self.X_df = rnaseq_filtered_df
+        self.y_df = y_filtered_df
+        self.gene_features = gene_features
+
+
+    def run_cv_for_cancer_type(self, gene, cancer_type, sample_info,
```

Comment:
Is it necessary to couple training models with data processing? If you decide you want to try two different ways of processing data down the line you'll have to either make another class or modify the training logic. If your cv function is separate from data processing and takes in a list of dataframes as an argument, then you can use it with any kind of preprocessing without needing to modify it

Reply:
To clarify, I agree that it makes sense to only load the data once, but you can load the data with this class and then pass it into separate training functions unless I'm missing something

Reply:
What you're saying makes sense to me, I guess I'm just having trouble picturing how it would work. So model training (currently in this function, `run_cv_for_cancer_type`) would be a separate class or function in a separate file, that just takes the data (self.X and self.y) from this class as arguments?

Maybe an example would be helpful - do you know of any examples that do what you're describing (either from your own code or from other repos)?

Reply:
Right, the function currently only uses the seed, `X_df`, and `y_df` from the class it lives in, along with a couple functions. You could write the same function with `X_df` and `y_df` as parameters and it wouldn't have to live in the class. You would then have to call `preprocess_data`, `save_results` etc from outside the class.

I guess the idea is that the concept of training and the concept of data management are two different things that don't really need to know about each other. The code as it currently is looks fine, it just might be harder to manage in the future since changes to data processing now might affect training

Reply:
Makes sense. I'll split this into another PR since it will be a decent sized change, but I agree it would be better to separate these two steps.